### PR TITLE
Rework Protocol Documentation

### PR DESCRIPTION
**Pull request by theaddonn**
This PR reworks large chunks of the protocol documentation. Please look over it and give it a few thoughts, I dont think it is finished yet, as some sections are still marked as _To be documented_.

Thats also why this is a draft right now...

### CODE REVIEW

**Comment by QuazChick** [comment]
Might be better to also include "Protocol" in the page titles so that if they're embedded somewhere like Discord the link will say "Bedrock Protocol" rather than just "Bedrock".

**Comment by theaddonn** [comment]
Got that and your prior suggestion on discord covered!

File: docs/servers/bedrock.md [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+  - Frames
+  - AnimationExpression
+- ArmSize
+- SkinColor
+- PersonaPieces = Array of:
+  - PackId
+  - PieceId
+  - IsDefault
+  - PieceType
+  - ProuctId
```
Comment (QuazChick) [inline docs/servers/bedrock.md]: Typo?
```suggestion
  - ProductId
```

Comment (theaddonn) [reply docs/servers/bedrock.md]: Thats a typo, but to my defense - It's a typo made by Mojang...

File: docs/servers/server-software.md [hunk 2]
```diff
@@ -22,198 +21,142 @@ mentions:
...
+    - bedrock-crustaceans
 description: Software for Bedrock servers.
 ---
 
-Minecraft servers allow players to play online or via a local area network with other people. This is very common within Java Edition Minecraft, but is also possible on Bedrock. [You can download the official Mojang BDS software here.](https://www.minecraft.net/en-us/download/server/bedrock).
+Minecraft servers allow players to play online or via a local area network with other people.
+[You can download the official Mojang BDS software here](https://www.minecraft.net/en-us/download/server/bedrock).
 
 Alongside the Vanilla BDS offering, many community projects exist, in a variety of languages.
+These other projects get seperated into Custom Server Software (Custom) and BDS-Based.
```
Comment (SuperLlama88888) [inline docs/servers/server-software.md]: separated*

Comment (theaddonn) [reply docs/servers/server-software.md]: Thanks pointing this out!

File: docs/servers/raknet.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+
+| Type                 | Size | Range           | Notes                                                               |
+|----------------------|------|-----------------|---------------------------------------------------------------------|
+| u8 (byte)            | 1    | 0-255           | A single Byte                                                       |
+| i16 (short)          | 2    | -32768 - 32767  | Signed 16-bit integer                                               |
+| u16 (unsigned short) | 2    | 0 - 65535       | Unsigned 16-bit integer                                             |
+| u24 (unsigned int24) | 3    | 0 - 2^24-1      | Unsigned 21-bit integer                                             |
+| i64 (long)           | 8    | -2^63 to 2^63-1 | Signed 64-bit integer                                               |
+| bool (boolean)       | 1    | 0 - 1           | `0` is `false`, while `1` is `true`                                 |
+| String               | N/A  |                 | A String prefixed by a unsigned short which depicts the length      |
```
Comment (ghost) [inline docs/servers/raknet.md]: I feel you should note the fact that the strings prefixed length is in Big Endian

Comment (theaddonn) [reply docs/servers/raknet.md]: should be covered now!

**Comment by theaddonn** [comment]
Since currently dont really have the time needed, Id still like this PR to be merged. Even though some areas are undocumented, do you think this could be merged?

**Comment by SirLich** [comment]
@theaddonn If you resolve conflicts I will merge.

**Comment by theaddonn** [comment]
@SirLich All conflicts are now resolved!

**Pull request merged**